### PR TITLE
feat(tabstops-auto-target-page-vis): SVG drawer draws visualizations for failure instances

### PR DESCRIPTION
**Pull request by waabid**
#### Details

SVG drawer will now use data coming in to draw different kinds of errors for tab stops.

##### Motivation

feature work.

##### Context

This work is still incomplete: there's some undesired behavior on when lines are connecting elements and when they are not. That will be tinkered with in a future PR.

Screen grab of what the visualizations look like below:
image

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/injected/visualization/svg-drawer.ts [hunk 1]
```diff
@@ -228,6 +236,7 @@ export class SVGDrawer extends BaseDrawer {
     }
 
     private createLinesInTabOrderVisualization(
+        items: TabbedItem[],
```
Comment (lisli1) [inline src/injected/visualization/svg-drawer.ts:239]: Are we adding this parameter to be clearer about which TabbedItem array we are using?

File: src/injected/visualization/svg-drawer.ts [hunk 2]
```diff
@@ -265,11 +274,44 @@ export class SVGDrawer extends BaseDrawer {
...
+
+        const circleConfiguration =
+            item.itemType === TabbedItemType.ErroredItem
+                ? drawerConfig.erroredCircle
+                : drawerConfig.missingCircle;
+
+        const newCircle = this.svgShapeFactory.createCircle(centerPosition, circleConfiguration);
+        const newLabel = this.svgShapeFactory.createTabIndexLabel(
+            centerPosition,
+            drawerConfig.tabIndexLabel,
```
Comment (sfoslund) [inline src/injected/visualization/svg-drawer.ts:299]: This doesn't necessarily need to be done in this PR, but eventually we will want to use a different config for error labels since we will want the text on them to be red. See the erroredTabIndexLabel config I added here: https://github.com/microsoft/accessibility-insights-web/pull/5199/files#diff-00053e82d4b379e30958fd1ca42842a8fadbd20d903faaa76bc26e1397656017R66

File: src/injected/visualization/svg-drawer.ts [hunk 3]
```diff
@@ -265,11 +274,44 @@ export class SVGDrawer extends BaseDrawer {
...
+            item.tabOrder,
+        );
+
+        const focusIndicator: FocusIndicator = {
+            circle: newCircle,
+            tabIndexLabel: newLabel,
```
Comment (sfoslund) [inline src/injected/visualization/svg-drawer.ts:305]: Again, doesn't have to be in this PR but we will want a failure label here too: https://github.com/microsoft/accessibility-insights-web/pull/5199/files#diff-da79de6a20c6af768f69e58a29187f44ccf98e439534ee29d54dffcf8265f27cR306-R316

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts [hunk 4]
```diff
@@ -55,16 +55,39 @@ describe('SVGDrawer', () => {
     }
 
     test('initialize', () => {
-        fakeDocument.body.innerHTML = "<div id='id1'></div>";
```
Comment (sfoslund) [inline src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:58]: It seems like we're mostly updating existing tests here. Are we going to add test coverage for the new logic?

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts [hunk 5]
```diff
@@ -701,14 +767,16 @@ describe('SVGDrawer', () => {
         drawerUtilsMock.verifyAll();
 
         expect(circles.length).toBe(2);
-        expect(lines.length).toBe(1);
+        expect(lines.length).toBe(0);
```
Comment (lisli1) [inline src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:770]: Confused why we expect 0 lines when the test says "with solid line"?

Comment (lisli1) [reply src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:770]: Is this related to the undesired behavior mentioned in the description?

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts [hunk 6]
```diff
@@ -772,9 +858,9 @@ describe('SVGDrawer', () => {
 
         drawerUtilsMock.verifyAll();
 
-        expect(circles.length).toBe(2);
-        expect(lines.length).toBe(0);
-        expect(labels.length).toBe(1);
+        expect(circles.length).toBe(3);
```
Comment (lisli1) [inline src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:861]: Does this count only tabbedElements that have `isFailure: true`?

File: src/injected/visualization/svg-drawer.ts (hunk 1 above)
Comment (waabid) [reply src/injected/visualization/svg-drawer.ts:239]: Yeah, technically this doesn't need to happen in this PR but I have a refactor that will make use of the fact that it is agnostic to which array is used.

File: src/injected/visualization/svg-drawer.ts (hunk 2 above)
Comment (waabid) [reply src/injected/visualization/svg-drawer.ts:299]: Good catch! I'll look into it!

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts (hunk 4 above)
Comment (waabid) [reply src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:58]: It's hard to parse because of how git shows the diffs but I removed a test (test was not necessary) and added a new test that contains failures and such specifically. All new code is covered (can be verified using `yarn watch:test --collect-coverage` and making a line deletion change to the file to cause it to run tests). There's some code in main that is not tested but not pertaining to this PR.

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts (hunk 6 above)
Comment (waabid) [reply src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:861]: No, these are all circles created.

In general: my personal opinion of this is that testing in this file is lackluster but since this needs a significant refactor and improving tests without that refactor would be work that would be undone/redone when refactored.

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts (hunk 5 above)
Comment (waabid) [reply src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:770]: I need to check on this again but this seems to be a part of the weird diff behavior: I didn't modify this test case but it makes it seem like I did because of the test deletion.

File: src/injected/visualization/svg-drawer.ts (hunk 2 above)
Comment (waabid) [reply src/injected/visualization/svg-drawer.ts]: Added!

File: src/tests/unit/tests/injected/visualization/svg-drawer.test.ts [hunk 7]
```diff
@@ -1077,6 +1084,11 @@ describe('SVGDrawer', () => {
         return labels;
     }
 
+    function findFailureLabels(): NodeListOf<Element> {
+        const circles = shadowContainer.querySelectorAll('.insights-svg-failure-label');
```
Comment (lisli1) [inline src/tests/unit/tests/injected/visualization/svg-drawer.test.ts:1088]: nit: should this be called labels instead of circles?

**Review by lisli1 — APPROVED** [review]
Small comment but LGTM

**Review by sfoslund — APPROVED** [review]
